docs(gatsby-plugin-react-helmet): add React hooks compatibility information (#12873)
gatsbyjs/gatsby: packages/gatsby-plugin-react-helmet/README.md

<Helmet title="foo bar" defer={false} />
```

## Compatibility with React 16.8 useEffect hook

> If you are using this plugin with React hooks, you may notice some errors that look like `maximum call stack size exceeded`. To ensure everything is running smoothly when using these technologies together, make sure of the following:

Reviewer: ```suggestion
If you are using this plugin with React hooks, you may notice some errors like `maximum call stack size exceeded`. To ensure everything is running smoothly when using these technologies together, make sure to validate the following:
```
